src/v/archival/ntp_archiver_service.cc

```diff
@@ -1018,7 +1029,8 @@ ntp_archiver::schedule_single_upload(const upload_context& upload_ctx) {
         .archiver_term = _start_term,
         .segment_term = upload.term,
         .delta_offset_end = delta_offset_end,
-        .sname_format = cloud_storage::segment_name_format::v2,
+        .sname_format = cloud_storage::segment_name_format::v3,
+        .metadata_size_hint = tx_size,
```

Comment:
BTW do we expect this to be used for anything beyond transactions? Wondering if we should name this tx_size_hint or something and reserve "metadata" for something more generic